Automattic/wp-calypso
# Simple payments block: use Formik for validating fields

#### Changes proposed in this Pull Request

Implement Simple payments block form validation using Formik.

In this PR we still rely on `props.setAttributes` but we'll refactor that out as a next step and then we can remove most of the custom `onChange`-handlers and rely on Formik's internal helper and read form values just before saving from `props.values`. 

#### Testing instructions

- In Gutenberg, insert simple payments button: https://calypso.live/gutenberg/post?branch=update/simple-payments-block-formik
- Test:
  - adding content,
  - removing content,
  - inserting invalid price (negatives, zero, non-numerals, leave it empty)
  - changing currency to HUF or JPY while the price has decimals: these currencies don't support decimals and you should now see the error right away.
